codalab/server/bundle_manager.py

```diff
@@ -692,6 +703,24 @@ def _get_staged_bundles_to_run(self):
                     bundle,
                     {'state': State.FAILED, 'metadata': {'failure_message': failure_message}},
                 )
+            elif bundle.metadata.request_queue:
+                # For those bundles that were requested to run on a private worker, if the private worker does not exist
+                # temporarily, we filter out those bundles so that they won't be dispatched to run on workers.
+                request_queue = bundle.metadata.request_queue
+                matched_workers = self._get_matched_workers(request_queue, workers)
+                if not matched_workers:
+                    # TODO: update bundles with batch operation: implement function batch_update_bundles()
+                    self._model.update_bundle(
```

Comment:
Seems like this is going to get executed a lot even though the `staged_status` doesn't change?  Not sure what to do about that though...

Reply:
That's a good point. We can cache those bundles. They shouldn't take too much space.

Reply:
I don't think you can cache because a bundle can eventually move out of this set.

Reply:
Added logic to handle this.